packages/vue/src/utils/create-ts-config.ts

```diff
@@ -1,7 +1,92 @@
 import { Tree, updateJson, writeJson } from '@nx/devkit';
 import * as shared from '@nx/js/src/utils/typescript/create-ts-config';
+import { isUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-setup';
 
 export function createTsConfig(
+  host: Tree,
+  projectRoot: string,
+  type: 'app' | 'lib',
+  options: {
+    strict?: boolean;
+    style?: string;
+    bundler?: string;
+    rootProject?: boolean;
+    unitTestRunner?: string;
+  },
+  relativePathToRootTsConfig: string
+) {
+  if (isUsingTsSolutionSetup(host)) {
+    createTsConfigForTsSolution(
+      host,
+      projectRoot,
+      type,
+      options,
+      relativePathToRootTsConfig
+    );
+  } else {
+    createTsConfigForNonTsSolution(
+      host,
+      projectRoot,
+      type,
+      options,
+      relativePathToRootTsConfig
+    );
+  }
+}
+
+export function createTsConfigForTsSolution(
+  host: Tree,
+  projectRoot: string,
+  type: 'app' | 'lib',
+  options: {
+    strict?: boolean;
+    style?: string;
+    rootProject?: boolean;
+    unitTestRunner?: string;
+  },
+  relativePathToRootTsConfig: string
+) {
+  const json = {
+    files: [],
+    include: [],
+    references: [
+      {
+        path: type === 'app' ? './tsconfig.app.json' : './tsconfig.lib.json',
+      },
+    ],
+  } as any;
+
+  // inline tsconfig.base.json into the project
+  if (options.rootProject) {
+    json.compileOnSave = false;
+    json.compilerOptions = {
+      ...shared.tsConfigBaseOptions,
+      ...json.compilerOptions,
+    };
+    json.exclude = ['node_modules', 'tmp'];
```

Comment:
We shouldn't inline the compiler options. Regardless of being a root project, the `tsconfig.json` file still extends from `tsconfig.base.json` and includes no files.